mkdocs/gh_deploy.py

```diff
@@ -5,19 +5,25 @@
 log = logging.getLogger(__name__)
 
 
-def gh_deploy(config):
+def gh_deploy(config, commit_message=None):
 
     if not os.path.exists('.git'):
         log.info('Cannot deploy - this directory does not appear to be a git '
                  'repository')
         return
 
-    log.info("Copying '%s' to `gh-pages` branch and pushing to GitHub.",
-             config['site_dir'])
+    command = ['ghp-import', '-p', config['site_dir']]
+
+    remote_branch = config.get('remote_branch', 'gh-pages')
+    command.extend(['-b', config['remote_branch']])
```

Comment:
Either this needs to be in an if statement (`if config['remote_branch'] is not None:`) or it needs to pass in the `remote_branch` var obtained in the previous line. That said, I don't see a `remote_branch` config setting defined in the [defaults][1] (which means it will never exist as unknown keys will raise an error).  I suspect this may have been a bug in the existing code as well (at least after 9d3594c landed).

And I wonder if this is the right place to define the default. Perhaps the default should be left to the ghp-import tool. Thus the `if not None` suggestion -- a branch name should only be passed if one is defined.

I notice that you do use the `remote_branch` var later in the log message. Perhaps rather than assuming the default is "gh-pages", when a branch is not explicitly defined the message should simply say: "Copying <site_dir> to the default branch..." rather than actually naming the branch. In fact, if the `remote_branch` var is only used for the log message, seems like it should be calculated there. That way future reviewers of the code won't wonder why it is not used when building the command.

[1]: https://github.com/mkdocs/mkdocs/blob/master/mkdocs/config/defaults.py